controllers/drplacementcontrol_controller.go

```diff
@@ -926,20 +926,28 @@ func (r *DRPlacementControlReconciler) getDRPolicy(ctx context.Context,
 func (r DRPlacementControlReconciler) addLabelsAndFinalizers(ctx context.Context,
 	drpc *rmn.DRPlacementControl, placementObj client.Object, log logr.Logger,
 ) error {
-	// add label and finalizer to DRPC
-	labelAdded := rmnutil.AddLabel(drpc, rmnutil.OCMBackupLabelKey, rmnutil.OCMBackupLabelValue)
-	finalizerAdded := rmnutil.AddFinalizer(drpc, DRPCFinalizer)
+	update := false
 
-	if labelAdded || finalizerAdded {
+	update = rmnutil.AddLabel(drpc, rmnutil.OCMBackupLabelKey, rmnutil.OCMBackupLabelValue)
+	update = update || rmnutil.AddFinalizer(drpc, DRPCFinalizer)
```

Comment:
If `update` is true then AddFinalizer will not be called. The next reconciliation will work but not the first time.

Reply:
Good catch, flipped the order